pkg/tcpip/stack/stack.go

```diff
@@ -832,9 +849,18 @@ func (s *Stack) GetRouteTable() []tcpip.Route {
 
 // AddRoute appends a route to the route table.
 func (s *Stack) AddRoute(route tcpip.Route) {
+	if !route.IsValid() {
+		panic(fmt.Sprintf("Route is not valid: %v", route))
+	}
 	s.route.mu.Lock()
 	defer s.route.mu.Unlock()
-	s.route.mu.table = append(s.route.mu.table, route)
+	// Sort routes in descending order based on the prefix length.
+	i := sort.Search(len(s.route.mu.table), func(j int) bool {
```

Comment:
I'm using a binary search and insertion since I think it's unnecessary to sort the table again when it's already sorted. If you have a very large routing table it probably can make a difference since it has a lower time complexity, O(log n) for binary search vs in best case O(n) for sorting.